crates/next-core/src/util.rs

```diff
@@ -66,3 +73,155 @@ pub async fn foreign_code_context_condition(next_config: NextConfigVc) -> Result
     };
     Ok(result)
 }
+
+#[derive(Default, PartialEq, Eq, Clone, Copy, Debug, TraceRawVcs, Serialize, Deserialize)]
+pub enum NextRuntime {
+    #[default]
+    NodeJs,
+    Edge,
+}
+
+#[turbo_tasks::value]
+#[derive(Default)]
+pub struct NextSourceConfig {
+    pub runtime: NextRuntime,
+}
+
+/// An issue that occurred while resolving the React Refresh runtime module.
+#[turbo_tasks::value(shared)]
+pub struct NextSourceConfigParsingIssue {
+    path: FileSystemPathVc,
+    detail: StringVc,
+}
+
+#[turbo_tasks::value_impl]
+impl Issue for NextSourceConfigParsingIssue {
+    #[turbo_tasks::function]
+    fn severity(&self) -> IssueSeverityVc {
+        IssueSeverity::Warning.into()
+    }
+
+    #[turbo_tasks::function]
+    fn title(&self) -> StringVc {
+        StringVc::cell("Unable to parse config export in source file".to_string())
+    }
+
+    #[turbo_tasks::function]
+    fn category(&self) -> StringVc {
+        StringVc::cell("parsing".to_string())
+    }
+
+    #[turbo_tasks::function]
+    fn context(&self) -> FileSystemPathVc {
+        self.path
+    }
+
+    #[turbo_tasks::function]
+    fn description(&self) -> StringVc {
+        StringVc::cell(
+            "The exported configuration object in a source file need to have a very specific \
+             format from which some properties can be statically parsed at compiled-time."
+                .to_string(),
+        )
+    }
+
+    #[turbo_tasks::function]
+    fn detail(&self) -> StringVc {
+        self.detail
+    }
+}
+
+#[turbo_tasks::function]
+pub async fn parse_config_from_source(module_asset: AssetVc) -> Result<NextSourceConfigVc> {
+    if let Some(ecmascript_asset) = EcmascriptModuleAssetVc::resolve_from(module_asset).await? {
+        if let ParseResult::Ok { program, .. } = &*ecmascript_asset.parse().await? {
+            if let Program::Module(module) = &program {
```

Comment:
Are scripts supported by Next?

Reply:
When it has an export it's not a script

Reply:
I understand, but that doesn't answer the question. Does next support `exports.handler = function () {}` CJS edge functions? Are devs forced to use ESM? Do we need to add a Script branch to this function?

Reply:
Afaik we only support config in that format:

```
export const config = {
  runtime: "edge"
};
```

Reply:
https://nextjs.org/docs/advanced-features/react-18/switchable-runtime#edge-api-routes

Reply:
next.js also looks for export declarations: https://github.com/vercel/next.js/blob/8f05c0c07e0796b6e4e8c3610a49e3828d1ee8ef/packages/next-swc/crates/core/src/page_config.rs#L74